app/src/main/java/com/github/sdp_begreen/begreen/fragments/SendPostFragment.kt

```diff
@@ -79,57 +83,66 @@ class SendPostFragment : Fragment() {
         shareBtn?.setOnClickListener {
 
             // fetch current user. He is necessarily not null
-            val user = connectedUserViewModel.currentUser.value!!
+            connectedUserViewModel.currentUser.value?.let{user ->
 
-            //create a metadata file
-            var metadata : TrashPhotoMetadata? = null
+                //create a metadata file
+                var metadata: TrashPhotoMetadata?
 
-            //fetch description on UI
-            view?.findViewById<TextInputEditText>(R.id.post_description).also {
+                //fetch description on UI
+                view?.findViewById<TextInputEditText>(R.id.post_description).also {
 
-                val caption = it?.text.toString()
+                    val caption = it?.text.toString()
 
-                // TODO : Let the user choose the category like what we did in googlemap for the bin category
-                var category = TrashCategory.ORGANIC
+                    // TODO : Let the user choose the category like what we did in googlemap for the bin category
+                    val category = TrashCategory.ORGANIC
 
-                /*
-                //fetch category on UI
-                view?.findViewById<TextInputEditText>(R.id.post_category)?.also { cat ->
-                    category = cat.text.toString()
-                }*/
+                    /*
+                    //fetch category on UI
+                    view?.findViewById<TextInputEditText>(R.id.post_category)?.also { cat ->
+                        category = cat.text.toString()
+                    }*/
 
-                metadata = TrashPhotoMetadata(null, ParcelableDate.now, user?.id, caption, category)
+                    metadata = TrashPhotoMetadata(null, ParcelableDate.now, user.id, caption, category)
 
+                }
+
+                // Update the user and return
+                lifecycleScope.launch {
+                    updateUser(metadata, user)
+                    if (!param_test) returnToCamera()
+                }
             }
+        }
+    }
 
-            // Post picture to firebase
-            lifecycleScope.launch {
-                view?.findViewById<ImageView>(R.id.preview)?.drawable?.toBitmap()?.let { bitmap ->
-                    // Get the stored metadata
-                    val storedMetadata = metadata?.let {
+    /**
+     * Helper function to update the user after posting a photo
+     */
+    private suspend fun updateUser(metadata: TrashPhotoMetadata?, user: User) {
+        view?.findViewById<ImageView>(R.id.preview)?.drawable?.toBitmap()?.let { bitmap ->
```

Comment:
Here I would rather use `also` instead of `let`, as the return value of the function is not used.

Reply:
That is more logical indeed, well found!